src/lib.rs

```diff
@@ -43,8 +44,8 @@ impl std::fmt::Display for MacAddressError {
         write!(
             f,
             "{}",
-            match self {
-                &InternalError => "Internal API error",
+            match *self {
+                InternalError => "Internal API error",
```

Comment:
Generally I prefer prefixing `&` before the terms instead of derefing the match value, it keeps it consistent for cases in which the value isn't `Copy`.